api/command.go

```diff
@@ -20,7 +20,7 @@ import (
 type CommandProvider interface {
 	GetTrigger() string
 	GetCommand(c *Context) *model.Command
-	DoCommand(c *Context, channelId string, message string) *model.CommandResponse
+	DoCommand(c *Context, post *model.Post, message string) *model.CommandResponse
```

Comment:
I love the idea of passing around a struct here instead of just a single value, but instead of using a `model.Post`, could you instead make a new struct called something like `model.CommandArgs` instead? It feels like we're using a `Post` for something that it wasn't intended since most of the commands only need the `ChannelId` and `RootId` set right now. Using a new struct will also allow us to add extra fields in the future if we ever decide that there's anything else we decide we want to pass into commands.

It could look something like this:
```go
struct {
    Command string `json:"command"`
    ChannelId string `json:"channel_id"`
    RootId string `json:"root_id"`
    // any future fields
}
```